### ISSUE TITLE
Add unleash_user resource and data blocks

### ISSUE THREAD

**Issue by deepbrook** [philips-labs/terraform-provider-unleash#31]
Ahoi!

It'd be quite neat if blocks for user management could be implemented. The relevant APIs exist afait (admin api), so this should be a straight-forward implementation.

The resource block would look something like this:

```hcl
resource "unleash_user" "example" {
  email     = "bob.joe@thismail.com"  // According to the API docs, either email or name are required 
  name      = "Bob Joe"
  rootRole  = "Editor"  // Alternatively, the API docs state the role's ID may be given instead.
  sendEmail = false
}
```

The `data` block could use either the get-user endpoint, or use the search endpoint:

```hcl
data "unleash_user" "example" {
  id = 1234
}

data "unleas_users" "search" {
  query = "bob"  // Docs state a minimum length of two characters, and this searches both email and name fields of users.
}
```
The latter block should exposition a collection of the matched users. I'm not sure if it's ok to run a sub-sequent API run based on the results here; running any matches against the get-user endpoint behind the scenes would ensure that the returned matches have the same attributes as the single `unleash_user` data block imported via ID.

It may also be a good idea to *not* use the `loginAttempts` field of the user object, as that may be a possible source for a false-positive while checking state drift.

### References
- [Admin API Docs](https://docs.getunleash.io/api/admin/user-admin)

**Comment by jdelucaa** [comment]
Hi @deepbrook 👋 

Thanks for suggesting this and for the detailed information, it makes it a lot easier for sure! I will see if I can find time this week to refine it a little bit more and start something.

**Comment by jdelucaa** [comment]
Available on `v0.2.0` of  the provider! 🎊

Some extra info: https://github.com/philips-labs/terraform-provider-unleash/pull/34#issuecomment-1156770303 

Please let me know if this works for you. Thanks!
